src/test/java/uk/nhs/hee/tis/trainee/credentials/service/CredemtoalEvemtServiceTest.java

```diff
@@ -0,0 +1,176 @@
+    placement2.setCredentialId(String.valueOf(PLACEMENT_2_credentialId));
+    placement2.setTraineeId(PLACEMENT_2_traineeId);
+    placement2.setCredentialType("TRAINING_PLACEMENT");
+
+    placement3.setIssuedAt(PLACEMENT_3_issuedAt);
+    placement3.setTisId(PLACEMENT_3_tisId);
+    placement3.setCredentialId(String.valueOf(PLACEMENT_3_credentialId));
+    placement3.setTraineeId(PLACEMENT_3_traineeId);
+    placement3.setCredentialType("TRAINING_PLACEMENT");
+
+    programme1.setIssuedAt(PROGRAMME_1_issuedAt);
+    programme1.setTisId(PROGRAMME_1_tisId);
+    programme1.setCredentialId(String.valueOf(PROGRAMME_1_credentialId));
+    programme1.setTraineeId(PROGRAMME_1_traineeId);
+    programme1.setCredentialType("TRAINING_PROGRAMME");
+
+    programme2.setIssuedAt(PROGRAMME_2_issuedAt);
+    programme2.setTisId(PROGRAMME_2_tisId);
+    programme2.setCredentialId(String.valueOf(PROGRAMME_2_credentialId));
+    programme2.setTraineeId(PROGRAMME_2_traineeId);
+    programme2.setCredentialType("TRAINING_PROGRAMME");
+  }
+
+  @Test
+  void getTheLatestPlacementCredential() {
+    CredentialType type = CredentialType.valueOf("TRAINING_PLACEMENT");
+    String token = TestJwtUtil.generateTokenForTisId("47165");
+
+    List<CredentialMetadata> allPlacementCredentialList = new ArrayList<>();
+    allPlacementCredentialList.add(placement1);
+    allPlacementCredentialList.add(placement2);
+    allPlacementCredentialList.add(placement3);
+
+    List<CredentialMetadata> filteredPlacementList = new ArrayList<>();
+    filteredPlacementList.add(placement1);
+    filteredPlacementList.add(placement3);
+
+    when(repository.findByCredentialTypeAndTraineeId(type.getIssuanceScope(), token))
+        .thenReturn(allPlacementCredentialList);
+
+    List<CredentialMetadata> latestCredentials =
+        credentialEventService.getTheLatestCredential(type, token);
+
+    verify(repository, times(1))
+        .findByCredentialTypeAndTraineeId(type.getIssuanceScope(), token);
+    assertEquals(filteredPlacementList, latestCredentials);
+  }
+
+  @Test
+  void getTheLatestProgrammeCredential() {
+    CredentialType type = CredentialType.valueOf("TRAINING_PROGRAMME");
+    String token = TestJwtUtil.generateTokenForTisId("47165");
+
+    List<CredentialMetadata> allProgrammeCredentialList = new ArrayList<>();
+    allProgrammeCredentialList.add(programme1);
+    allProgrammeCredentialList.add(programme2);
+
+    List<CredentialMetadata> filteredProgrammeList = new ArrayList<>();
+    filteredProgrammeList.add(programme1);
+
+    when(repository.findByCredentialTypeAndTraineeId(type.getIssuanceScope(),
+        token)).thenReturn(allProgrammeCredentialList);
+
+    List<CredentialMetadata> latestCredentials =
+        credentialEventService.getTheLatestCredential(type, token);
+
+    verify(repository, times(1))
+        .findByCredentialTypeAndTraineeId(type.getIssuanceScope(), token);
+    assertEquals(filteredProgrammeList, latestCredentials);
```

Comment:
I prefer not to build an compare an "expectedResult" type object. Instead we can check the size and contents of the result, which makes it much easier to read the expectations of the test later.

```java
assertThat("Unexpected credential count", latestCredentials.size(), is(1));
assertThat("Unexpected credential", latestCredentials, hasItem(programme1));
```